src/math/exp_fcn_hifi4.c

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+#include <sof/math/exp_fcn_hifi4.h>
+#include <sof/common.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#ifdef EXPONENTIAL_HIFI4
+#include <xtensa/tie/xt_hifi4.h>
+
+
+static void mul_s64(ae_int64 in_0, ae_int64 in_1, ae_int64 * ptroutbitshi,
+		    ae_int64 * ptroutbitslo);
+
+static int64_t mul_s64_sr_sat_near(int64_t a, int64_t b)
+{
+	ae_int64 result;
+	ae_int64 u64_chi;
+	ae_int64 u64_clo;
+
+	mul_s64(a, b, &u64_chi, &u64_clo);
+
+	ae_int64 roundup = AE_AND64(u64_clo, BIT_MASK_LOW_Q27P5);
+
+	roundup = AE_SRLI64(roundup, 27);
+	ae_int64 temp = AE_OR64(AE_SLAI64(u64_chi, 36), AE_SRLI64(u64_clo, 28));
+
+	u64_clo = AE_ADD64(temp, roundup);
+
+	result = u64_clo;
+
+	return result;
+}
+
+/*
+ * Arguments	: int64_t in_0
+ *		  int64_t in_1
+ *		  uint64_t *ptroutbitshi
+ *		  uint64_t *ptroutbitslo
+ * Return Type	: void
+ */
+static void mul_s64(ae_int64 in_0, ae_int64 in_1, ae_int64 *ptroutbitshi, ae_int64 *ptroutbitslo)
+{
+	ae_int64 producthihi, producthilo, productlolo;
+
+	ae_int32x2 in0_32 = AE_MOVINT32X2_FROMINT64(in_0);
+	ae_int32x2 in1_32 = AE_MOVINT32X2_FROMINT64(in_1);
+
+	ae_ep ep_lolo = AE_MOVEA(0);
```

Comment:
This seems to make it not compatible with hifi3. Did yo consider making such version?